src/main/java/com/epam/izh/rd/online/entity/Book.java

```diff
@@ -16,5 +19,54 @@
  * 6) Переопределить метод toString с выводом всех полей (не забывайте alt+inset)
  */
 public abstract class Book {
+    private int numberOfPages;
+    private String name;
+
+    public Book() {
+    }
+
+    public Book(int numberOfPages, String name) {
+        this.numberOfPages = numberOfPages;
+        this.name = name;
+    }
+
+    public int getNumberOfPages() {
+        return numberOfPages;
+    }
+
+    public void setNumberOfPages(int numberOfPages) {
+        this.numberOfPages = numberOfPages;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Book book = (Book) o;
+
+        return new EqualsBuilder().append(numberOfPages, book.numberOfPages).append(name, book.name).isEquals();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17, 37).append(numberOfPages).append(name).toHashCode();
+    }
 
+    @Override
+    public String toString() {
+        return "Book{" +
```

Comment:
ToStringBuilder.reflectionToString(this, ToStringStyle.JSON_STYLE)?

Reply:
didn't know this. Thank you.